x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/group_stats_renderers.tsx

```diff
@@ -0,0 +1,129 @@
+  {
+    defaultMessage: 'Multi',
+  }
+);
+
+export const INTEGRATION_ICON_TEST_ID = 'alert-summary-table-integration-cell-renderer-icon';
+export const INTEGRATION_LOADING_TEST_ID = 'alert-summary-table-integration-cell-renderer-loading';
+
+interface IntegrationProps {
+  /**
+   * Aggregation buckets for integrations
+   */
+  signalRuleIdBucket: GenericBuckets;
+}
+
+/**
+ * Renders the icon for the integration that matches the rule id.
+ * In AI for SOC, we can retrieve the integration/package that matches a specific rule, via the related_integrations field on the rule.
+ */
+export const Integration = memo(({ signalRuleIdBucket }: IntegrationProps) => {
+  const signalRuleId = signalRuleIdBucket.key;
+  const { integration, isLoading } = useGetIntegrationFromRuleId({ ruleId: signalRuleId });
+
+  return (
+    <EuiSkeletonText data-test-subj={INTEGRATION_LOADING_TEST_ID} isLoading={isLoading} lines={1}>
+      {integration ? (
+        <CardIcon
+          data-test-subj={INTEGRATION_ICON_TEST_ID}
+          icons={integration.icons}
+          integrationName={integration.title}
+          packageName={integration.name}
+          size="s"
+          version={integration.version}
+        />
+      ) : null}
+    </EuiSkeletonText>
+  );
+});
+Integration.displayName = 'Integration';
+
+/**
+ * Return a renderer for integration aggregation.
+ */
+export const getIntegrationComponent = (
+  bucket: RawBucket<AlertsGroupingAggregation>
+): GroupStatsItem[] => {
+  const signalRuleIds = bucket.signalRuleIdSubAggregation?.buckets;
+
+  if (!signalRuleIds || signalRuleIds.length === 0) {
+    return [];
+  }
+
+  if (signalRuleIds.length === 1) {
+    return [
+      {
+        title: STATS_GROUP_SIGNAL_RULE_ID,
+        component: <Integration signalRuleIdBucket={signalRuleIds[0]} />,
+      },
+    ];
+  }
+
+  return [
+    {
+      title: STATS_GROUP_SIGNAL_RULE_ID,
+      component: <>{STATS_GROUP_SIGNAL_RULE_ID_MULTI}</>,
+    },
+  ];
+};
+
+/**
+ * Returns stats to be used in the`extraAction` property of the EuiAccordion component used within the kbn-grouping package.
+ * It handles custom renders for the following fields:
+ * - signal.rule.id
+ * - kibana.alert.severity
+ * - kibana.alert.rule.name
+ * And returns a default view for all the other fields.
+ *
+ * These go hand in hand with groupingOptions, groupTitleRenderers and groupStatsAggregations.
+ */
+export const groupStatsRenderer = (
+  selectedGroup: string,
+  bucket: RawBucket<AlertsGroupingAggregation>
+): GroupStatsItem[] => {
+  const defaultBadges: GroupStatsItem[] = DEFAULT_GROUP_STATS_RENDERER(selectedGroup, bucket);
+  const severityComponent: GroupStatsItem[] = getSeverityComponent(bucket);
+  const integrationComponent: GroupStatsItem[] = getIntegrationComponent(bucket);
+
+  switch (selectedGroup) {
+    case 'signal.rule.id':
+      return [...severityComponent, getRulesBadge(bucket), ...defaultBadges];
```

Comment:
nit, you can pull out `getRulesBadge(bucket)` the same way you did severity and integration

Reply:
true!

Reply:
fixed in https://github.com/elastic/kibana/pull/216744/commits/c4cb1b7e72deba56f9fa335ee27825f0d17ad899